docs: transforms updates (#69037)
metabase/metabase: docs/data-studio/transforms/transforms-overview.md

Once you change the transform's query or script, the next transform run (manual or scheduled) will use the updated query and write the results into the target table. If you have questions that query the transform's target table, they might break. For example, if your new transform query no longer includes a column that a downstream question was relying on, that question will break.

### Edit transform's target

_Data Studio > Transforms > Settings_

To edit transform's target table, i.e., the table where the query results are written, click on "Change target". You'll need to select whether you want to keep the old target table, or delete it. Deletion can't be undone.

**Questions built on the old target will _not_ be transferred to the new target table.** If you delete the old target table, any questions using the old transform target table will break. If you keep the old target around, the questions built on it won't break but they will _not_ use the new target table, and so will become outdated.

## Run a transform

_Data Studio > Transforms > Run_

> You can run a transform manually or on schedule (e.g., hourly).

Reviewer: ```suggestion
You can run a transform manually or schedule the transform using tags and jobs.
```

To run a transform manually, visit the transform in **Data Studio > Transforms > Runs** and click **Run**.

To schedule a transform, you'll need to assign one or more tags to it, then create a [scheduled job](./jobs-and-runs.md) that picks up those tags.

You can see the time and status of the latest transform run on the transform's page, or in the [Runs view](./jobs-and-runs.md). The time of the run is given in the system's timezone.

For Python transforms, you'll also see the transform's execution logs.

## Transform dependencies

> Transform dependency graph is only available on Pro and Enterprise plans (both self-hosted and Metabase Cloud)

_Data Studio > Transforms > Dependencies_

Transform queries can use the data from other transforms, and query-based transforms can also reference Metabase questions and models. For example, you can have a transform that uses data from a `raw_events` table and writes to a `stg_events` table, and then create another transform that uses data from the `stg_events` table and writes to an `events` table. Metabase.

> Metabase will track transform dependencies on other transforms, and execute transforms in reasonable order. So for example, if transform that creates the `events` table relies on the `stg_events` table which, in turn, is also created by a transform, then whenever you run transform generating `events`, the transform for `stg_events` will run as well.

Reviewer: ```suggestion
Metabase will track transform dependencies and execute transforms in a reasonable order. So for example, if transform B relies on a table created by transform A, Metabase will run transform A first, then run transform B.
```

If a job includes a transform that depends on a table created by another transform, then the job will run all the tagged transforms and their dependencies, even if they lack tags, see [Jobs and runs](jobs-and-runs.md) for more information.

### Incremental transforms

_Data Studio > Transforms > Settings_

Incremental transforms allow you to only write the _new_ data to the database when the transform is executed. For example, you might have new transaction data coming in every day. You can Metabase transforms to only write data for transactions that haven't been processed before

### Prerequisites for incremental transforms

- There is a column in your data that Metabase can check for new values to determine which data is new. We'll refer to this as a "Checkpoint" column.
- The checkpoint column has to be either numeric or timestamp, and has to have increasing values. Metabase will determine what "new" data is by looking for values that are _greater than_ already-written checkpoint values. usually this will be some kind of "ID" or "Created At" column.
- The checkpoint column should be present in both input and output table.

> - Your schema is stable, meaning that the structure of the tables is not going to change from run to run

Reviewer: ```suggestion
- Your schema is stable, meaning that the structure of the tables is not going to change from run to run.
```
